yapio.c

```diff
@@ -1251,9 +1268,36 @@ yapio_alloc_buffers(const yapio_test_group_t *ytg)
         yapio_destroy_buffers_and_abort();
 }
 
+/* read input md file to initialize yapioMyRank's md array */
+static void
+yapio_initialize_source_md_buffer_from_file(const yapio_test_group_t *ytg)
```

Comment:
you do not check that you have the completeness of md file accross all processes. What happen if one of them does not have its md file to read ?

I would add a collective opeation that would sum the number of process that have read successfully its md file.
i would see somewhere an MPI_Allreduce (..... MPI_SUM ....) somewhere, and each process would put 1 in the buffer.

Reply:
adding it to the bottom of the function